File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max_profit = 0;
        int buy = prices[0];
        int cost = 0;
        for(int i=1; i<prices.length; i++) {
            cost = prices[i] - buy;
            max_profit = Math.max(max_profit, cost);
            buy = Math.min(prices[i], buy);
        }
        return max_profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        buy = prices[0]
        sell = 0

        for i in range(len(prices)):
            sell = prices[i] - buy
            max_profit = max(max_profit, sell)
            buy = min(buy, prices[i])

        return max_profit